joinmarket/blockchaininterface.py

```diff
@@ -115,6 +115,341 @@ def estimate_fee_per_kb(self, N):
+                    continue
+                # If there are addresses in the chunk with unspent outputs, get data
+                for addr in req_block:
+                    res = self.make_bci_request(bcinfo_url + addr)
+                    count = count + 1
+                    if res == False:
+                        continue
+                    data = json.loads(res)['unspent_outputs']
+
+                    # for each unspent output
+                    for uo in data:
+                        wallet.unspent[uo['tx_hash_big_endian'] + ':' + str(uo['tx_output_n'])] = {
+                            'address' : addr,
+                            'value' : int(uo['value'])}
+        for u in wallet.spent_utxos:
+            wallet.unspent.pop(u, None)
+
+        self.last_sync_unspent = time.time()
+        log.debug('blockchaininfo sync_unspent took ' +
+                        str((self.last_sync_unspent - st)) + 'sec')
+
+    def add_tx_notify(self, txd, unconfirmfun, confirmfun, notifyaddr):
+        unconfirm_timeout = 10 * 60  # seconds
+        unconfirm_poll_period = 15
+        confirm_timeout = 2 * 60 * 60
+        confirm_poll_period = 5 * 60
+
+        class NotifyThread(threading.Thread):
+
+            def __init__(self, txd, unconfirmfun, confirmfun):
+                threading.Thread.__init__(self)
+                self.daemon = True
+                self.unconfirmfun = unconfirmfun
+                self.confirmfun = confirmfun
+                self.tx_output_set = set([(sv['script'], sv['value'])
+                                          for sv in txd['outs']])
+                self.output_addresses = [
+                    btc.script_to_address(scrval[0], get_p2pk_vbyte())
+                    for scrval in self.tx_output_set]
+                log.debug('txoutset=' + pprint.pformat(self.tx_output_set))
+                log.debug('outaddrs=' + ','.join(self.output_addresses))
+
+            def run(self):
+                st = int(time.time())
+                unconfirmed_txid = None
+                unconfirmed_txhex = None
+                while not unconfirmed_txid:
+                    time.sleep(unconfirm_poll_period)
+                    if int(time.time()) - st > unconfirm_timeout:
+                        log.debug('checking for unconfirmed tx timed out')
+                        return
+                    bcinfo_url = 'https://blockchain.info/unspent?active='
+                    random.shuffle(self.output_addresses) # seriously weird bug with blockr.io
+                    # might not be needed for blockchain.info
+                    
+                    # create list for requests of all addresses
+                    data_list = []
+                    #for each of the output addresses, request data and add it to list
+                    for output_address in self.output_addresses:
+                        res = BlockchaininfoInterface().make_bci_request(bcinfo_url + output_address)
```

Comment:
This creates and discards an entire object every time this look is run. Maybe it's worth putting `make_bci_request` outside the BlockchaininfoInterface class